### ISSUE TITLE
BUG: Anthropic api not working at all, "Invalid bearer token" every time.

### ISSUE THREAD

**Issue by soundslikeinfo** [obsidianmd/obsidian-clipper#243]
**Version (please complete the following information):**

- OS: macOS 15.1.1
- Browser Firefox & Safari
- Web Clipper version: 0.10.4
- Obsidian version: n/a

**Describe the bug**
I've inserted my Anthropic keys into many GPT clients, and have created fresh keys in this instance. No matter what I do, I can not get Claude Sonnet Models to work with Obsidian Web Clipper in Firefox or Safari. Does anyone else have this same issue?

The different anthropic models I've tried, are:
Model ID: claude-3-5-sonnet-latest
Model ID: claude-3-5-sonnet-20241022
Model ID: claude-3-5-haiku-latest
Model ID: claude-3-5-haiku-20241022
Model ID: claude-3-opus-20240229
Model ID: claude-3-opus-latest
Model ID: claude-3-sonnet-20240229

I receive this error everytime for Anthropic:
```
Anthropic error:  {"type":"error","error":{"type":"authentication_error","message":"Invalid bearer token"}}
```

Everytime using Anthropic as the interpreter:
![41860](https://github.com/user-attachments/assets/6556ee27-c52a-451f-93ca-a3a323a59f3d)

**Expected behavior**
What did you expect to happen?
OpenAI keys work every time, as shown below:
Everytime using OpenAI as the interpreter:
![86032](https://github.com/user-attachments/assets/e6a64680-9489-4cd4-849d-74c64060237b)

**URLs where the bug occurs**
Any specific web pages where the bug can be replicated.
Website that I tried clipping with Anthropic model, for example: https://nadia.xyz/ideas

1. Go to Obsidian Web Clipper Settings
2. Click on Interpreter underneath settings
3. Add provider: Anthropic
4. Add Base URL: https://api.anthropic.com/v1/messages, and API Key "sk-ant-api03-...."
5. Add a new Model for Anthropic & Model ID = claude-3-5-sonnet-20241022
6. Visit website (ie. https://nadia.xyz/ideas) and open the webclipper.
7. Choose interpreter for Anthropic and Interpret

**Your template file**

If you are using a custom template (i.e. not the Default template). Go to Web Clipper settings and click **More** → **Copy as JSON**. Paste the JSON code below.

```json
[web-interpreter-clipper.json](https://github.com/user-attachments/files/18056098/web-interpreter-clipper.json)
```

**Comment by kepano** [comment]
It sounds like the error is with Anthropic. You should check your API token, or generate a new one.

**Comment by soundslikeinfo** [comment]
> It sounds like the error is with Anthropic. You should check your API token, or generate a new one.

I've checked and created new keys, and assigned them to workspaces that I know are working in other apps. It is still giving me the same issue. My OpenAI workflows are working in the meantime. I wonder if anyone has gotten their Anthropic interpreters to work.

**Comment by kepano** [comment]
I am not able to reproduce the issue. It all works okay on my end. The error is directly from Anthropic so I would assume that's where the error is from.

**Comment by soundslikeinfo** [comment]
Obsidian Web Clipper v0.10.3 for Safari is not working with Anthropic. Here is what I gathered.
I used the web inspector for the Obsidian Web Clipper.

1. For **Anthropic**, the Authorization header should be:
x-api-key: <your-api-key>
_Not Authorization: Bearer <api-key>_
**Source:**
[source](https://docs.anthropic.com/en/api/messages) **x-api-key**, **anthropic-version**
![44894](https://github.com/user-attachments/assets/0870c7a1-ad17-458b-87fc-2eb1d7492b96)

_Etc._
The incorrect authorization header sent from Safari's Obsidian Web Clipper Extension:
<img width="1252" alt="image" src="https://github.com/user-attachments/assets/76d493d7-36c6-42a1-97cc-22d5fc80e7bc">

**Comment by soundslikeinfo** [comment]
I looked at [interpreter.ts](https://github.com/obsidianmd/obsidian-clipper/blob/4d00d63fc739cbe19c6744cc757124c1687df00c/src/utils/interpreter.ts) and it looks like the correct headers are in place. So I checked if I was somehow using OpenAI still as the provider for the Anthropic models, and nothing looked wrong on my end. I think the conditional isn't flagging truthy for 'anthropic' and always reverting to the default 'OpenAI' for the headers.

**Comment by fabioscarsi** [comment]
> **Version (please complete the following information):**
> 
> * OS: macOS 15.1.1
> * Browser Firefox & Safari
> * Web Clipper version: 0.10.4
> * Obsidian version: n/a
> 
> **Describe the bug** I've inserted my Anthropic keys into many GPT clients, and have created fresh keys in this instance. No matter what I do, I can not get Claude Sonnet Models to work with Obsidian Web Clipper in Firefox or Safari. Does anyone else have this same issue?
> 
> The different anthropic models I've tried, are: Model ID: claude-3-5-sonnet-latest Model ID: claude-3-5-sonnet-20241022 Model ID: claude-3-5-haiku-latest Model ID: claude-3-5-haiku-20241022 Model ID: claude-3-opus-20240229 Model ID: claude-3-opus-latest Model ID: claude-3-sonnet-20240229
> 
> I receive this error everytime for Anthropic:
> 
> ```
> Anthropic error:  {"type":"error","error":{"type":"authentication_error","message":"Invalid bearer token"}}
> ```
> 
> ![](https://storage.googleapis.com/pieces-web-extensions-cdn/pieces-light.png)
> Copy And Save
> ![](https://storage.googleapis.com/pieces-web-extensions-cdn/link-light.png)
> Share
> ![](https://storage.googleapis.com/pieces-web-extensions-cdn/copilot_light.png)
> Ask Copilot
> ![](https://storage.googleapis.com/pieces-web-extensions-cdn/settings_light.png)
> Everytime using Anthropic as the interpreter: ![41860](https://private-user-images.githubusercontent.com/353860/393692234-6556ee27-c52a-451f-93ca-a3a323a59f3d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NDI5MzAsIm5iZiI6MTczMzc0MjYzMCwicGF0aCI6Ii8zNTM4NjAvMzkzNjkyMjM0LTY1NTZlZTI3LWM1MmEtNDUxZi05M2NhLWEzYTMyM2E1OWYzZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwOVQxMTEwMzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNDc5NGRjYWJhYWE4NzNhNDgyYzMyZDY2OTQ0YTgyZWJiY2NiNDFmY2ZiNTY1OTQ2YWE4NDlmYmMzMGQ1MTllJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4WtYnUA2lbYgaDosapCLhrnxo8ecQlWgRjgQbcO8prQ)
> 
> **Expected behavior** What did you expect to happen? OpenAI keys work every time, as shown below: Everytime using OpenAI as the interpreter: ![86032](https://private-user-images.githubusercontent.com/353860/393692340-e6a64680-9489-4cd4-849d-74c64060237b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NDI5MzAsIm5iZiI6MTczMzc0MjYzMCwicGF0aCI6Ii8zNTM4NjAvMzkzNjkyMzQwLWU2YTY0NjgwLTk0ODktNGNkNC04NDlkLTc0YzY0MDYwMjM3Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwOVQxMTEwMzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yYjc0ZDRmNDcxMzQwMDBjZGRlNDM4ZTJiMTBlOWUxY2IwYTY5OThhMmM3ZWU1NGRkM2MwOGYyYjA4YWM4MzM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Iwshngt64pXMDiVT2rZKR3a9txYl0VDGQawXsjYRm_I)
> 
> **URLs where the bug occurs** Any specific web pages where the bug can be replicated. Website that I tried clipping with Anthropic model, for example: https://nadia.xyz/ideas
> 
> Steps to reproduce the behavior:
> 
> 1. Go to Obsidian Web Clipper Settings
> 2. Click on Interpreter underneath settings
> 3. Add provider: Anthropic
> 4. Add Base URL: https://api.anthropic.com/v1/messages, and API Key "sk-ant-api03-...."
> 5. Add a new Model for Anthropic & Model ID = claude-3-5-sonnet-20241022
> 6. Visit website (ie. https://nadia.xyz/ideas) and open the webclipper.
> 7. Choose interpreter for Anthropic and Interpret
> 
> **Your template file**
> 
> If you are using a custom template (i.e. not the Default template). Go to Web Clipper settings and click **More** → **Copy as JSON**. Paste the JSON code below.
> 
> [web-interpreter-clipper.json](https://github.com/user-attachments/files/18056098/web-interpreter-clipper.json)

I have the same problem. After the last two updates of Obsidian Web Clipper, I can't make anthropic models work, even though they were working before.
I, my case, MacBook Pro, Apple silicon, MacOS Sequoia 15.1.1, Chrome, and Obsidian Web Clipper 0.10.3. The Api Key is the same that was working a week ago using Obsidian Web Clipper with anthropic models set as custom.

**Comment by kepano** [comment]
Fixed in 0.10.5
